### DISCUSSION TITLE
Testing feedback

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#1168] [Feature feedback]
Let's use this space to provide feedback for testing. 

We'd like to find out:

 - what do you want to test?
 - what testing helpers you require?

**Comment by leewaa** [comment]
It would be really really helpful have documentation on how to fill-in/select dates from the datepicker in a system test.

**Reply by adrianthedev** [reply]
Thanks for writing, @leewaa. We know that's a pain and we introduced a lot of testing helpers in the upcoming version 3.
We don't have a backport for Avo 2, but I can share with you the helpers here and you can add them to your project.

```ruby
    def set_picker_day(date)
      find(".flatpickr-day[aria-label=\"#{date}\"]").click
      sleep 0.2
    end

    def set_picker_hour(value)
      find(".flatpickr-hour").set(value)
    end

    def set_picker_minute(value)
      find(".flatpickr-minute").set(value)
    end

    def set_picker_second(value)
      find(".flatpickr-second").set(value)
    end

    def open_picker
      text_input.click
    end

    def close_picker
      find('[data-target="title"]').click
      sleep 0.3
    end
```

**Reply by leewaa** [reply]
That helps a lot. Thank you!
